identity_account_storage/src/storage/stronghold.rs

```diff
@@ -81,8 +82,9 @@ impl Storage for Stronghold {
 
     let public_key: PublicKey = retrieve_public_key(&tmp_client, &tmp_location)?;
 
-    let did: IotaDID = IotaDID::new_with_network(public_key.as_ref(), network)
-      .map_err(|err| crate::Error::DIDCreationError(err.to_string()))?;
+    let did: CoreDID = IotaDID::new_with_network(public_key.as_ref(), network)
+      .map_err(|err| crate::Error::DIDCreationError(err.to_string()))?
+      .into();
```

Comment:
We should probably introduce an enum `DIDType` or something like that, that can be passed to `Storage::did_create` to tell the implementation what flavor of DID to generate. That enum only needs to hold an `Iota` variant for now, but that way it can be more easily extended to support other DID types.